platform/core/src/capabilities/PersistenceCapability.js

```diff
@@ -148,13 +148,19 @@ define(
          */
         PersistenceCapability.prototype.refresh = function () {
             var domainObject = this.domainObject;
+            var $q = this.$q;
 
             // Update a domain object's model upon refresh
             function updateModel(model) {
-                var modified = model.modified;
-                return domainObject.useCapability("mutation", function () {
-                    return model;
-                }, modified);
+                if (model === undefined) {
```

Comment:
Ah, yeah. I think that this is because the [`PersistenceAggregator`](https://github.com/nasa/openmct/blob/f07934f5127d4bbcf84601a11a66bb1d232a630c/platform/persistence/aggregator/src/PersistenceAggregator.js#L52) doesn't aggregate new style object providers. Any object that is exposed via a new style object provider, but retrieved via an old style persistence provider will currently return undefined. I just added an explicit promise rejection in this case rather than returning undefined. The `RefreshingController` where it's called from remains unchanged, so in effect ignores the rejected callback. This was a convenience fix to get the demo working without errors, but doesn't address the heart of the issue.